src/sinks/http.rs

```diff
@@ -392,6 +394,14 @@ impl util::http::HttpSink for HttpSink {
                 w.write_all(&body).expect("Writing to Vec can't fail");
                 body = w.finish().expect("Writing to Vec can't fail").into_inner();
             }
+            Compression::Zstd(level) => {
```

Comment:
Looks like there is precedent in the unit tests for covering specific compression settings, I think it would be nice to add a case to cover this one.

Reply:
Always appreciate tests, but it looks like only `gzip` is getting tested today?

The duplication in this `match` also strikes me as something we could rewrite to be less repetitive.

Reply:
> Always appreciate tests, but it looks like only gzip is getting tested today?

🤷 never hurts :)

> The duplication in this match also strikes me as something we could rewrite to be less repetitive.

Agreed.

Reply:
done.